unit-tests/post-processing/test-post-processing-from-bag.py

```diff
@@ -0,0 +1,201 @@
+            frameset.keep()
+            sensor_to_framesets[sensor_name] = frameset
+            all_paris = True
+        else:
+            all_paris = False
+
+
+def check_condition(sensors):
+    with frames_lock:
+        return len(sensor_to_frame) >= len(sensors)
+
+
+def wait_for_condition(sensors):
+    while not check_condition(sensors):
+        time.sleep(0.0001)
+    completion_event.set()
+
+
+def callback_for_get_frames(frame, sensor_to_frame, sensors):
+    with frames_lock:
+        if len(sensor_to_frame) < len(sensors):
+            frame.keep()
+            sensor_name = rs.sensor_from_frame(frame).get_info(rs.camera_info.name)
+            sensor_to_frame[sensor_name] = frame
+
+
+def get_frames(sensors):
+    frames = []
+    for s in sensors:
+        s.open(s.get_stream_profiles())
+
+    for s in sensors:
+        s.start(lambda frame: callback_for_get_frames(frame, sensor_to_frame, sensors))
+
+    timeout = 30
+    start_time = time.time()
+
+    condition_thread = threading.Thread(target=wait_for_condition(sensors))
+    condition_thread.start()
+    condition_thread.join(timeout)
+
+    test.check(time.time() - start_time < timeout)
+
+    for s in sensors:
+        s.stop()
+
+    for s in sensors:
+        s.close()
+
+    for sf in sensor_to_frame:
+        frames.append(sensor_to_frame[sf])
+
+    return frames
+
+
+def get_composite_frames(sensors):
+    frame_processor = rs.processing_block(process_frame)
+    for sensor in sensors:
+        sensor.open(sensor.get_stream_profiles())
+
+    for sensor in sensors:
+        sensor.start(lambda frame: (frame.keep(), frame_processor.invoke(frame)))
+
+    while True:
+        with frame_processor_lock:
+            if len(sensor_to_framesets) >= len(sensors) and all_paris is True:
+                break
+            time.sleep(0.0001)
+
+    for sf in sensor_to_framesets:
+        composite_frames.append(sensor_to_framesets[sf])
+
+    for sensor in sensors:
+        sensor.stop()
+
+    for sensor in sensors:
+        sensor.close()
+
+    return composite_frames
+
+
+def compare_processed_frames_vs_recorded_frames(record_block, file):
+    temp_dir = os.path.join(tempfile.gettempdir())
+    filename = os.path.join(temp_dir, "all_combinations_depth_color.bag")
+
+    ctx = rs.context()
+    dev = ctx.load_device(filename)
+    dev.set_real_time(False)
+
+    sensors = dev.query_sensors()
+    frames = get_composite_frames(sensors)
+    ref_dev = ctx.load_device(os.path.join(temp_dir, file))
+    ref_dev.set_real_time(False)
+
+    ref_sensors = ref_dev.query_sensors()
+    ref_frames = get_frames(ref_sensors)
+
+    test.check_equal(len(ref_frames), len(frames))
+
+    for i in range(len(frames)):
+        fs_res = record_block.process(frames[i])
+        validate_ppf_results(fs_res, ref_frames[i])
+
+
+################################################################################################
+test.start("Test align depth to color from recording")
```

Comment:
I suggest you use `test.closure` as it's a little more robust, and will automatically catch exceptions etc.
Look at `live/options/test-presets.py` for an example.